jruby-ext/src/main/java/com/purbon/jrmonitor/monitors/ProcessMonitor.java

```diff
@@ -2,71 +2,127 @@
 
 import com.sun.management.UnixOperatingSystemMXBean;
 
-import javax.management.MBeanServer;
 import java.lang.management.ManagementFactory;
 import java.lang.management.OperatingSystemMXBean;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Objects;
 
 /**
  * Created by andrewvc on 5/12/16.
  */
 public class ProcessMonitor {
     private static final OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();
-    private static final MBeanServer platformMxBean = ManagementFactory.getPlatformMBeanServer();
 
-    public class Report {
-        private long memTotalVirtualInBytes = -1;
-        private short cpuSystemPercent = -1;
-        private short cpuProcessPercent = -1;
-        private long cpuMillisTotal = -1;
-        private boolean isUnix;
-        private long openFds = -1;
-        private long maxFds = -1;
-
-        private Map<String, Object> map = new HashMap<>();
+    class Reporter {
+        public Long getOpenFileDescriptorCount() {
+            return Long.valueOf(-1);
+        }
+        public Long getMaxFileDescriptorCount() {
+            return Long.valueOf(-1);
+        }
+        public Long getProcessCpuTime() {
+            return Long.valueOf(-1);
+        }
+        public double getProcessCpuLoad() {
+            return Long.valueOf(-1);
+        }
+        public double getSystemCpuLoad() {
+            return Long.valueOf(-1);
+        }
 
-        public Report() {
-            this.isUnix = osMxBean instanceof UnixOperatingSystemMXBean;
+        public Long getCommittedVirtualMemorySize() {
+            return Long.valueOf(-1);
+        }
+    }
 
-            // Defaults are -1
-            if (this.isUnix) {
-                UnixOperatingSystemMXBean unixOsBean = (UnixOperatingSystemMXBean) osMxBean;;
+    class UnixReporter extends Reporter {
+        private final UnixOperatingSystemMXBean osMxBean;
 
-                this.openFds = unixOsBean.getOpenFileDescriptorCount();
-                this.maxFds =  unixOsBean.getMaxFileDescriptorCount();
+        public UnixReporter(UnixOperatingSystemMXBean osMxBean) {
+            this.osMxBean = osMxBean;
+        }
 
-                this.cpuMillisTotal = unixOsBean.getProcessCpuTime();
-                this.cpuProcessPercent = scaleLoadToPercent(unixOsBean.getProcessCpuLoad());
-                this.cpuSystemPercent = scaleLoadToPercent(unixOsBean.getSystemCpuLoad());
+        public Long getOpenFileDescriptorCount() {
+            try {
+                return this.osMxBean.getOpenFileDescriptorCount();
+            } catch(Throwable e) {
+                return -1L;
+            }
+        }
+        public Long getMaxFileDescriptorCount() {
+            try {
+                return this.osMxBean.getMaxFileDescriptorCount();
+            } catch(Throwable e) {
```

Comment:
Why are we catching top level throwables? Is there not a more specific class we should be catching?